icu4j/main/classes/core/src/com/ibm/icu/impl/units/UnitConverter.java

```diff
@@ -153,9 +153,9 @@ private static Factor processFactorWithoutDivision(String factorWithoutDivision)
         }
 
         /**
-         * Clone this <code>Factor</code>.
+         * Copy this <code>Factor</code>.
          */
-        protected Factor clone() {
```

Comment:
If method name is `clone()`, it overrides `Object.clone()`. Returning covariant type in subclass is OK since Java 5.

When you implement clone() method, you should start with `super.clone()` which creates a new instance of the object in optimized way. Calling a constructor of the class is off from the convention. For this specific implementation, the end result will be equivalent, but such implementation does not work if parent is not `Object`.

If you really want to override `Object#clone()`, you should add `implements Cloneable` to the class. Again, this is not a strict contract, but it's out of standard convention. Static code analysis tool like Spotbugs tend to produce a warning.

I could change this method to standard `clone()` implementation, and add `implements Cloneable`. But after checking call site, changing the method name to `copy()` would be the less impact change.

Reply:
Thanks for the explanation/education!